### PR TITLE
Convert js to ts

### PR DESCRIPTION
**Pull request by huayunh**
Changes proposed in this Pull Request:
- Added type `UIColorType` for `ui/` and `BrandingColorType` for `branding/`.
- Used `tsconfig.json` from pxblue/themes.
- Removed all babel references in `package.json`.
- Updated `buildTest.sh`, `yarn.lock` accordingly.

### CODE REVIEW

File: branding/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "./dist/index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "build": "yarn && babel src --presets babel-preset-env --out-dir dist"
+    "build": "tsc"
```
Comment (joebochill) [inline branding/package.json]:
```suggestion
    "build": "yarn && tsc"
```

Comment (joebochill) [reply branding/package.json]: Let's keep the yarn command here to make sure dependencies are installed before the build

File: branding/src/palette.ts [hunk 2]
```diff
@@ -6,7 +6,20 @@ All rights reserved.
 This code is licensed under the BSD-3 license found in the LICENSE file in the root directory of this source tree and at https://opensource.org/licenses/BSD-3-Clause.
 **/
 
-export const blue = {
+type BrandingColorType = {
```
Comment (joebochill) [inline branding/src/palette.ts]: Can we unify BrandingColorType and UIColorType with a single definition (just make the missing fields optional ones)...ideally they are the same type.

File: package.json [hunk 3]
```diff
@@ -2,11 +2,11 @@
   "name": "PXBlueColors",
   "version": "2.0.0",
   "description": "PX Blue colors for Eaton applications",
-  "main": "index.js",
+  "main": "index.ts",
   "scripts": {
     "test": "./buildTest.sh",
-    "build:colors": "cd ui && yarn build && cd ..",
-    "build:colors-branding": "cd branding && yarn build && cd ..",
+    "build:colors": "cd ui && yarn && tsc && cd ..",
```
Comment (joebochill) [inline package.json]:
```suggestion
    "build:colors": "cd ui && yarn build && cd ..",
```

File: package.json [hunk 4]
```diff
@@ -2,11 +2,11 @@
...
+    "build:colors-branding": "cd branding && yarn && tsc && cd ..",
```
Comment (joebochill) [inline package.json]:
```suggestion
    "build:colors-branding": "cd branding && yarn build && cd ..",
```

Comment (joebochill) [reply package.json]: the build command (updated) will do yarn install and tsc, so we can leave it as is

File: package.json [hunk 5]
```diff
@@ -2,11 +2,11 @@
   "name": "PXBlueColors",
   "version": "2.0.0",
```
Comment (joebochill) [inline package.json]: We can bump this to a new major version 3.0.0

File: ui/package.json [hunk 6]
```diff
@@ -5,7 +5,7 @@
   "main": "./dist/index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "build": "yarn && babel src --presets babel-preset-env --out-dir dist"
+    "build": "tsc"
```
Comment (joebochill) [inline ui/package.json]:
```suggestion
    "build": "yarn && tsc"
```

File: ui/src/palette.ts [hunk 7]
```diff
@@ -6,7 +6,25 @@ All rights reserved.
...
+    500: string,
+    600: string,
+    700: string,
+    800: string,
+    900: string,
+    A100?: string,
+    A200?: string,
+    A400?: string,
+    A700?: string,
+    'contrastDefaultColor': string
```
Comment (joebochill) [inline ui/src/palette.ts]: I don't think 'contrastDefaultColor' needs to be quoted in these objects

File: branding/src/palette.ts [hunk 8]
```diff
@@ -186,7 +189,7 @@ export const goldenrod = {
     800: '#f0a920',
     900: '#ee9b14'
 };
-export const trophy = {
+export const trophy  = {
```
Comment (joebochill) [inline branding/src/palette.ts]: trophy is missing a Type

File: buildTest.sh [hunk 9]
```diff
@@ -6,21 +6,30 @@ NC='\033[0m' # No Color
 
 echo "Checking for required files..."
 echo -ne "  UI index: "
-if [ ! -f ./ui/dist/index.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
+if [ ! -f ./ui/dist/ui/src/index.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
```
Comment (joebochill) [inline buildTest.sh]: we should not be changing the path where the compiled components go...the final results should be in /dist, not two folders below that. There shouldn't be a /src folder under a /dist folder.

**Comment by huayunh** [comment]
Note to the two commits above: 
* the first commit flattens everything, but is using `amd` in tsconfig.json. This will cause some subsequent troubles (modules not recognized when testing with pxb:themes/react, file hierarchy "implicitly" referred inside the file)
* the second commit reverts everything back to the hierarchy version but specifies the entry point of the package in package.json

File: branding/package.json [hunk 10]
```diff
@@ -2,10 +2,11 @@
   "name": "@pxblue/colors-branding",
   "version": "2.0.2",
   "description": "Eaton Branding colors for Eaton applications",
-  "main": "./dist/index.js",
+  "main": "./dist/ndex.js",
```
Comment (joebochill) [inline branding/package.json]: typo here

File: branding/package.json [hunk 11]
```diff
@@ -2,10 +2,11 @@
   "name": "@pxblue/colors-branding",
   "version": "2.0.2",
```
Comment (joebochill) [inline branding/package.json]: can you update the version to 3.0.0

File: branding/readme.md [hunk 12]
```diff
@@ -35,7 +35,7 @@ yarn add @pxblue/colors-branding
 ## Usage
 Incorporating these colors into your project is handled differently depending on the framework that you are using.
 
-### Angular or NativeScript
+### Angular or Ionic
```
Comment (joebochill) [inline branding/readme.md]: Good catch

File: ui/package.json [hunk 13]
```diff
@@ -3,9 +3,10 @@
   "version": "1.0.13",
```
Comment (joebochill) [inline ui/package.json]: can we update this to 2.0.0

**Pull request merged**